Documentation/ceph-object-store-crd.md

```diff
@@ -162,3 +162,34 @@ The endpoint health check procedure is the following:
 6. Update CR health status check
 
 Rook-Ceph always keeps the bucket and the user for the health check, it just does a PUT and GET of an s3 object since creating a bucket is an expensive operation.
+
+## Security settings
+
+Ceph RGW supports encryption via KMS using HashiCorp Vault. Refer to the [vault kms section](ceph-cluster-crd.md#vault-kms) for detailed explanation.
+If these settings are defined, then RGW establish a connection between Vault and whenever S3 client sends a request with Server Side Encryption,
+it encrypts that using the key specified by the client. For more details w.r.t RGW, please refer [Ceph Vault documentation](https://docs.ceph.com/en/latest/radosgw/vault/)
+
+The `security` section contains settings related to KMS encryption of the RGW.
+
+```yaml
+security:
+  kms:
+    # name of the k8s config map containing all the kms connection details
+    connectionDetails:
+      KMS_PROVIDER: vault
+      VAULT_ADDR: http://vault.default.svc.cluster.local:8200
+      VAULT_BACKEND_PATH: rgw
+      VAULT_SECRET_ENGINE: kv
+      VAULT_BACKEND: v2
+    # name of the k8s secret containing the kms authentication token
+    tokenSecretName: rgw-vault-token
+```
+
+For RGW, please note the following:
+
+* `VAULT_SECRET_ENGINE` option is specifically for RGW to mention about the secret engine which can be used, currently supports two: [kv](https://www.vaultproject.io/docs/secrets/kv) and [transit](https://www.vaultproject.io/docs/secrets/transit). And for kv engine only version 2 is supported.
+* The Storage administrator needs to create a secret in the Vault server so that S3 clients use that key for encryption
+$ vault kv put rook/i<mybucketkey> key=$(openssl rand -base64 32) # kv engine
+$ vault write -f transit/keys/<mybucketkey> exportable=true # transit engine
```

Comment:
What does `exportable=true` do?

Reply:
@leseb: AFAIR that's how we can enable key in transit engine, we don't need to give key-value pair like in kv engine copied from https://docs.ceph.com/en/octopus/radosgw/vault/

Reply:
That does not really tell me why we need `exportable=true` though.

Reply:
@leseb : https://www.vaultproject.io/api/secret/transit#exportable, If I understand correctly key will be visible to the user otherwise it will be hidden if it is set and in this example the user can read the key using `vault read transit/export/encryption-key/mybucketkey/1`